### PR TITLE
Docs Improvements (largely around TimeLock)

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:

* Fix various issues in TimeLock documentation (and in other places which may have changed in behaviour because of timelock, like the CLIs)
  - Update Manual Migration docs for Cassandra which were previously flawed
  - Update CLI Docs
  - Explain how to configure servers to use HTTP2
  - Add a section explaining how we use Paxos
* Fix unintuitive formatting in Transaction Protocol documentation

**Implementation Description (bullets)**:

* Added documentation to address the problems above.

**Concerns (what feedback would you like?)**:

* Nothing in particular. There is a bit of repetition at the end of the CLIs file, perhaps.

**Where should we start reviewing?**:

* Anywhere

**Priority (whenever / two weeks / yesterday)**:

* Whenever

### CODE REVIEW

File: docs/source/services/timelock_service/paxos.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+=====
+
+Paxos is an algorithm for *distributed consensus* - that is, getting servers to agree on a value that some server
+*proposes*, in the presence of communication and server failures. Its guarantees include:
+
+#. Safety: All servers will agree on the same value.
+#. Nontriviality: The value must have been *proposed* by at least one of the servers.
+#. Liveness: If a majority of servers is up and able to communicate, then the protocol can make progress.
+
+If the Timelock Server is configured to use Paxos, it will actually use separate instances of Paxos to agree on the
```
Comment (rhero) [inline docs/source/services/timelock_service/paxos.rst]: Can the timelock server be configured to use something other than Paxos? I'd probably remove this paragraph and instead combine it with the second with a slightly different focus. It's slightly hard to parse what we use Paxos for right now. 

This page could be presented with this kind of outline:

- The Paxos Algorithm: basically your high level description, link to the paper.
- How we use Paxos
  - Leadership election (primarily motivated for lock service)
  - async paxos (motivated for timestamp service)

File: docs/source/services/timelock_service/manual-migration.rst [hunk 2]
```diff
@@ -92,16 +95,21 @@ The steps for invalidating the old AtlasDB timestamp will vary, depending on you
...
+  This can be done easily using ``cqlsh``. The timestamp table is stored in the same keyspace that your
   AtlasDB client uses for its key-value service.
 
      .. code:: bash
 
         SELECT * FROM atlasdb."_timestamp";
         <note the value returned by this - call this K>
-        INSERT INTO atlasdb."_timestamp" (key, column1, column2, value) VALUES (0x7472, 0x7472, -1, K);
-        INSERT INTO atlasdb."_timestamp" (key, column1, column2, value) VALUES (0x7473, 0x7473, -1, 0x);
+        INSERT INTO atlasdb."_timestamp" (key, column1, column2, value) VALUES (0x7473, 0x7472, -1, K);
```
Comment (jeremyk-91) [inline docs/source/services/timelock_service/manual-migration.rst]: To be fair I think when this was written I didn't even know that lightweight transactions require the same partition key. Now I do :) thanks!

File: docs/source/services/timelock_service/paxos.rst (hunk 1 above)
Comment (jeremyk-91) [inline docs/source/services/timelock_service/paxos.rst]: Ah, okay. Yeah we don't officially support anything else at the moment.

Yep, that describes what I was going for here (the more interesting part is how we use Paxos, really). I'll reorganise this to fit that outline.

**Comment by rhero** [comment]
I responded to my own comments. Happy to merge this if you're cool with it @jeremyk-91.
